### PR TITLE
Add haproxy 1.8-rc1

### PR DESCRIPTION
**Pull request by stevenklar**

### CODE REVIEW

**Comment by cirocosta** [comment]
Hey @stevenklar , that looks pretty good 👍  By the way, there's already `rc2`.

**Comment by yosifkit** [comment]
Seems sane to me, but can we drop it in a `1.8-rc` folder?  That's what we usually do for pre-release builds in our other images.

File: 1.8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+FROM debian:jessie-backports
```
Comment (TimWolla) [inline 1.8/Dockerfile]: Should we update to stretch?

File: 1.8/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN apt-get update \
+	&& apt-get install -y --no-install-recommends \
+		liblua5.3-0 \
+		libpcre3 \
+		libssl1.0.0 \
```
Comment (TimWolla) [inline 1.8/Dockerfile]: If we do we should switch to `libssl1.1` here.

File: 1.8/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# see http://sources.debian.net/src/haproxy/jessie/debian/rules/ for some helpful navigation of the possible "make" arguments
+RUN set -x \
+	\
+	&& buildDeps=' \
+		gcc \
+		libc6-dev \
+		liblua5.3-dev \
+		libpcre3-dev \
+		libssl-dev \
```
Comment (TimWolla) [inline 1.8/Dockerfile]: And add `zlib1g-dev` here.

File: 1.8/alpine/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+# first arg is `-f` or `--some-option`
+if [ "${1#-}" != "$1" ]; then
+	set -- haproxy "$@"
+fi
+
+if [ "$1" = 'haproxy' ]; then
+	# if the user wants "haproxy", let's use "haproxy-systemd-wrapper" instead so we can have proper reloadability implemented by upstream
+	shift # "haproxy"
+	set -- "$(which haproxy-systemd-wrapper)" -p /run/haproxy.pid "$@"
```
Comment (TimWolla) [inline 1.8/alpine/docker-entrypoint.sh]: `haproxy-systemd-wrapper` was removed with haproxy 1.8.

**Comment by TimWolla** [comment]
I also recommend from staying away from haproxy 1.8 at this point.

I was about to create a pull request myself yesterday and ran into this ugly crasher while testing whether everything works: https://www.mail-archive.com/haproxy@formilux.org/msg27789.html.

File: 1.8/Dockerfile (hunk 1 above)
Comment (stevenklar) [reply 1.8/Dockerfile]: Good thought. Had the same in mind.

File: 1.8/Dockerfile (hunk 3 above)
Comment (stevenklar) [reply 1.8/Dockerfile]: Does HAProxy use this lib for compression?

**Comment by stevenklar** [comment]
Thought it's useful to provide the RC image for people who want to try and search/report bugs.

Keep this PR open until 1.8 is released fully would be fine I guess?

**Pull request closed without merging**